tests/transports/testquic.nim

```diff
@@ -424,3 +349,93 @@ suite "Quic transport":
     await runClient(server)
     await serverFut
     await server.stop()
+
+  asyncTest "multiaddress validation - accept valid addresses":
+    let transport = await createTransport()
+
+    let validAddresses =
+      @[
+        "/ip4/127.0.0.1/udp/1234/quic-v1", "/ip6/::1/udp/1234/quic-v1",
+        "/dns/example.com/udp/1234/quic-v1",
+      ]
+
+    for address in validAddresses:
+      check transport.handles(MultiAddress.init(address).tryGet())
+
+  asyncTest "multiaddress validation - reject invalid addresses":
+    let transport = await createTransport()
+
+    let invalidAddresses =
+      @[
+        "/ip4/127.0.0.1/udp/1234", # UDP without quic-v1
+        "/ip4/127.0.0.1/tcp/1234/quic-v1", # Wrong transport (TCP instead of UDP)
+        "/ip4/127.0.0.1/udp/1234/quic", # Legacy quic (not quic-v1)
+      ]
+
+    for address in invalidAddresses:
+      check not transport.handles(MultiAddress.init(address).tryGet())
+
+  asyncTest "address normalization - port assignment":
+    # Start with port 0 and verify it gets assigned a real port
+    let transport = await createTransport()
+    let ma = MultiAddress.init("/ip4/127.0.0.1/udp/0/quic-v1").tryGet()
+    await transport.start(@[ma])
+    defer:
+      await transport.stop()
```

Comment:
technically `createTransport(isServer = true)` does exactly this, however iam fine with repeating it